game-core/src/main/java/games/strategy/triplea/ui/RouteDescription.java

```diff
@@ -53,7 +53,7 @@ public boolean equals(final Object o) {
     diffX *= diffX;
     int diffY = end.y - other.end.y;
     diffY *= diffY;
-    final int endDiff = (int) Math.sqrt(diffX + diffY);
+    final int endDiff = (int) Math.sqrt((double) diffX + diffY);
```

Comment:
Ok, this is really picky here, I mean I get why sonarqube complains, but I don't think this will ever become an issue, nobody expects diffX + diffY to be out of range for an int, but in range of a float.
That's a good example why I'm not 100% agreeing with sonarqube everytime

Reply:
> That's a good example why I'm not 100% agreeing with sonarqube everytime

There is tuning to be done to downgrade or upgrade various alarms to account for false positives. On the other topic, arguing with static analysis IMO only goes so far. While yes, we can say we know more about the code than analysis and frankly are "smarter", on the other hand there is something to be said about code that is simple enough and valid enough that it passes any/all static analysis. 

While TripleA is not safety critical, the following was an interesting read and the practice I think follows for most any software project to minimize defects:

> The Power of Ten – Rules for Developing Safety Critical Code
> NASA/JPL Laboratory for Reliable Software 

> 10.Rule: All code must be compiled, from the first day of development, with allcompiler warnings enabled at the compiler’s most pedantic setting. All code must compile with these setting without any warnings. All code must be checked daily with at least one, but preferably more than one, state-of-the-art static source code analyzer and should pass the analyses with zero warnings.  

http://pixelscommander.com/wp-content/uploads/2014/12/P10.pdf